File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/format/YoutubeTrackFormatExtractor.java
package com.sedmelluq.discord.lavaplayer.source.youtube.format;

import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeSignatureResolver;
import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeTrackFormat;
import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeTrackJsonData;
import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;

import java.util.List;

public interface YoutubeTrackFormatExtractor {
    String DEFAULT_SIGNATURE_KEY = "signature";

    List<YoutubeTrackFormat> extract(
        YoutubeTrackJsonData response,
        HttpInterface httpInterface,
        YoutubeSignatureResolver signatureResolver
    );
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/format/YoutubeTrackFormatExtractor.kt
package com.sedmelluq.discord.lavaplayer.source.youtube.format

import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeSignatureResolver
import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeTrackFormat
import com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeTrackJsonData
import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface

interface YoutubeTrackFormatExtractor {
    companion object {
        const val DEFAULT_SIGNATURE_KEY = "signature"
    }

    fun extract(
        response: YoutubeTrackJsonData,
        httpInterface: HttpInterface,
        signatureResolver: YoutubeSignatureResolver
    ): List<YoutubeTrackFormat>
}
